concurrent.lib.sh

```diff
@@ -315,6 +315,7 @@ concurrent() (
             while read -r meta; do
                 printf "meta:%d:%s\n" "${1}" "${meta}" >> "${4}"
             done
+            exit $PIPESTATUS
```

Comment:
Done. Might be worth adding some basic tests to catch this kind of stuff in the future.